File: listview/src/main/java/com/melayer/listview/MainActivity.java
package com.melayer.listview;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    ListView listOfTechnologies;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        listOfTechnologies=findViewById(R.id.listOfTechnologies);
        showList();
    }


    void showList()
    {

        String[] technologies={"C",
                "Cpp",
                "Java",
                "Android",
                "HTML",
                "CSS","JS","AngularJs","Hadoop","C",
                "Cpp",
                "Java",
                "Android",
                "HTML",
                "CSS","JS","AngularJs","Hadoop"};

        ArrayAdapter adapter=new ArrayAdapter(this,R.layout.single_technology,R.id.textView,technologies);
        listOfTechnologies.setAdapter(adapter);

        listOfTechnologies.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
               // Toast.makeText(MainActivity.this, ""+position, Toast.LENGTH_SHORT).show();
                startActivity(new Intent(MainActivity.this,SecondActivity.class));
            }
        });


    }

}

File: listinkotlin/src/main/java/com/melayer/listinkotlin/MainActivity.kt
package com.melayer.listinkotlin

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.AdapterView
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        showList()
    }

    private fun showList() {

        val technologies = arrayOf("C", "Cpp", "Java", "Android", "HTML", "CSS", "JS", "AngularJs", "Hadoop", "C", "Cpp", "Java", "Android", "HTML", "CSS", "JS", "AngularJs", "Hadoop")
        val adapter = ArrayAdapter(this, R.layout.single_technology, R.id.textView, technologies)
        listOfTechnologies.adapter=adapter

        listOfTechnologies.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            // Toast.makeText(MainActivity.this, ""+position, Toast.LENGTH_SHORT).show();
            startActivity(Intent(this@MainActivity, SecondActivity::class.java))
        }
    }

}
